### PR TITLE
Attaching owner references to jobs create via "--from=cronjob/xyz"

### PR DESCRIPTION
**Pull request by odino**
/kind bug

**What this PR does / why we need it**:

Jobs created with `kubectl create job --from=cronjob/xyz` do not have owner references to the original cronjob. This is fairly tricky as the kubernetes dashboard and [other kind of dashboards](https://github.com/pietervogelaar/kubernetes-job-monitor) rely on them to group jobs together.

The scenario is very simple: an organization sets up cronjobs to run at a scheduled time. When one of the jobs breaks, developers might trigger a manual run. When they do, they will never be able to link the manual run back to the original cronjob.

Hopefully I made myself clear but, if not, I'm happy to try to explain this a little better :)

**Special notes for your reviewer**:

Not adding ownerReferences might have been done on purpose since this job is triggered manually via kubectl. If that's the case, I can alternatively add a flag to the command, something like `kubectl create job --from=cronjob/xyz --preserve-owner-ref`. Let me know what you think!

Original commit: https://github.com/kubernetes/kubernetes/commit/5b57c2db0067c8a16681fe9ec57ace3ad0342c8f

Ping to @erhudy @soltysh who were the original brains behind the "create from cronjob" feature :)

```release-note
NONE
```

### CODE REVIEW

**Comment by odino** [comment]
I had forgotten the tests since they didnt seem to be running on my machine, ouch :)

The 2nd commit should have fixed the tests (tested with `make test WHAT=./pkg/kubectl/cmd/create`), let me know your thoughts.

**Comment by odino** [comment]
Just FYI this has been tested on a local k8s cluster as well -- here's how it looks like:

image

What I ran was: 

```
$ ./_output/bin/kubectl create job --from=cronjob/hello this-was-triggered-manually
$ ./_output/bin/kubectl create job --from=cronjob/hello this-was-triggered-manually-again
$ ./_output/bin/kubectl create job --from=cronjob/hello this-was-triggered-manually-a-third-time
```

File: pkg/kubectl/cmd/create/create_job.go [hunk 1]
```diff
@@ -240,4 +242,10 @@ func (o *CreateJobOptions) createJobFromCronJob(cronJob *batchv1beta1.CronJob) *
 		},
 		Spec: cronJob.Spec.JobTemplate.Spec,
 	}
+
+	job.SetOwnerReferences([]metav1.OwnerReference{
+		*metav1.NewControllerRef(cronJob, apps.SchemeGroupVersion.WithKind("CronJob")),
```
Comment (soltysh) [inline pkg/kubectl/cmd/create/create_job.go]: Set it directly in `ObjectMeta` similarly how it's done in https://github.com/kubernetes/kubernetes/blob/2d956389dc57fb1715d1832b2d22c042de59bd1d/pkg/controller/cronjob/utils.go#L158-L165

**Review by soltysh — CHANGES_REQUESTED** [review]
Fix the nit, and please squash your changes. The idea itself is good, I don't see any reasons why jobs created from cronjobs should not be tied to their owners, which is useful when removing cronjob. 
/lgtm cancel

**Comment by odino** [comment]
@soltysh thanks for the hint.

I squashed and moved things around -- can you double check?

File: pkg/kubectl/cmd/create/create_job.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ import (
 
 	"github.com/spf13/cobra"
 
+	apps "k8s.io/api/apps/v1"
```
Comment (soltysh) [inline pkg/kubectl/cmd/create/create_job.go]: `appsv1` is the preferred convention.

File: pkg/kubectl/cmd/create/create_job_test.go [hunk 3]
```diff
@@ -184,6 +185,7 @@ func TestCreateJobFromCronJob(t *testing.T) {
 				Name: jobName,
 			}
 			job := o.createJobFromCronJob(tc.from)
+			tc.expected.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(tc.from, apps.SchemeGroupVersion.WithKind("CronJob"))}
```
Comment (soltysh) [inline pkg/kubectl/cmd/create/create_job_test.go]: Set the expected object as it should look like in appropriate test cases.

**Review by soltysh — CHANGES_REQUESTED** [review]
Some more comments.

File: pkg/kubectl/cmd/create/create_job_test.go (hunk 3 above)
Comment (odino) [reply pkg/kubectl/cmd/create/create_job_test.go]: TBH I'm having a hard time doing that without modifying the whole test case (unless I'm missing something). The test cases are a map of cronjob / job and I'm not aware of a way I can easily reference the cronjob from the job. I've extracted the cronjob outside of the map of test cases, but I can't say I like this more :)

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by odino** [comment]
@juanvallejo @soltysh any pointer on how to fix the bazel build errors?

@fejta I think the bot might want to chill / use exponential backoff after X failures :)

**Comment by odino** [comment]
I've updated `pkg/kubectl/cmd/create/BUILD` through `./hack/update-bazel.sh`

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by soltysh** [comment]
> @fejta I think the bot might want to chill / use exponential backoff after X failures :)

That's what `/hold` is for ;)

**Comment by odino** [comment]
@soltysh I can't seem to be able to see why the test is failing on `pull-kubernetes-e2e-gce-100-performance`, would you be able to help?

https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/73737/pull-kubernetes-e2e-gce-100-performance/34466/
https://gubernator.k8s.io/pr/73737

**Comment by odino** [comment]
@soltysh any help here? :)

**Comment by odino** [comment]
@soltysh @juanvallejo any help? :)

**Comment by sshishov** [comment]
Hello, I have a question here. When I create the `job` from `cronJob`, then it has the reference, but it lacks some fields: `blockOwnerDeletion: true` and `controller: true`. Due to this, the manual jobs are not deleted automatically according to `successfulJobsHistoryLimit` and `failedJobsHistoryLimit` settings. Could you please clarify how properly create **managed** manual Job from CronJob?

**Pull request merged**